src/filters.rs

```diff
@@ -34,22 +36,8 @@ pub struct LogFilter {
 
 impl LogFilter {
     fn matches(&self, log: &Log, latest_block_number: U64) -> bool {
-        let from = match self.from_block {
-            BlockNumber::Finalized
-            | BlockNumber::Pending
-            | BlockNumber::Committed
-            | BlockNumber::Latest => latest_block_number,
-            BlockNumber::Earliest => U64::zero(),
-            BlockNumber::Number(n) => n,
-        };
-        let to = match self.to_block {
-            BlockNumber::Finalized
-            | BlockNumber::Pending
-            | BlockNumber::Committed
-            | BlockNumber::Latest => latest_block_number,
-            BlockNumber::Earliest => U64::zero(),
-            BlockNumber::Number(n) => n,
-        };
+        let from = utils::to_real_block_number(self.from_block, latest_block_number);
```

Comment:
We should also update `get_block_transaction_count_by_number` and `get_block_by_number` in `node.rs`.

For `get_block_by_number`, it would be a slight refactor by removing a bunch of code:
```rust
let real_block_number = utils::to_real_block_number(block_number, &reader.current_miniblock);

reader
    .block_hashes
    .get(real_block_number)
    .and_then(|hash| reader.blocks.get(hash))
    .cloned()
    .or_else(|| {
        reader
            .fork_storage
            .inner
            .read()
            .expect("failed reading fork storage")
            .fork
            .as_ref()
            .and_then(|fork| {
                fork.fork_source
                    .get_block_by_number(block_number, true)
                    .ok()
                    .flatten()
            })
    })